File: app/lib/src/models/alert.dart
class Alert {
  final int id;
  final String title;
  final String content;
  final String type;
  final String notificationType;
  final String? department;
  final bool isActive;
  final DateTime createdAt;

  Alert({
    required this.id,
    required this.title,
    required this.content,
    required this.type,
    required this.notificationType,
    this.department,
    required this.isActive,
    required this.createdAt,
  });

  factory Alert.fromJson(Map<String, dynamic> json) {
    return Alert(
      id: json['id'],
      title: json['title'],
      content: json['content'],
      type: json['type'],
      notificationType: json['notification_type'],
      department: json['department'],
      isActive: json['is_active'],
      createdAt: DateTime.parse(json['created_at']),
    );
  }
} 

File: backend/app/Models/Alert.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;

class Alert extends Model
{
    use HasFactory;

    protected $fillable = [
        'title',
        'content',
        'type',
        'notification_type',
        'department',
        'reported_by',
        'is_active',
        'sent_at',
    ];

    protected $casts = [
        'is_active' => 'boolean',
        'sent_at' => 'datetime',
    ];

    public function reporter()
    {
        return $this->belongsTo(User::class, 'reported_by');
    }
} 